backend/src/main/java/com/eyetrade/backend/service/CurrencyRecordService.java

```diff
@@ -62,13 +70,18 @@ public CurrencyRecord getCurrencyRecord() throws IOException {
 
     public CurrencyRecord updateIfCurrenciesExpiredAndGetLastRecord(){
         CurrencyRecord record= currencyRepository.findLastRecord();
-        if(record==null || checkExpired(record.getTimestamp())){
-            try {
-                record=getCurrencyRecord();
-            } catch (IOException e) {
-                throw new RuntimeException(CURRENCIES_COULD_NOT_BE_UPDATED);
+        try{
+            if(record==null || checkExpired((new SimpleDateFormat(DB_DATE_TIME_FORMAT).parse(record.getDate())).getTime())){
```

Comment:
I think we should update the records with a scheduled job, instead of expiration checks. Because the users will need everyday data, even if they have not requested before.

Reply:
I think that this part is critical but if you want we can update it later.

Reply:
Actually I have added scheluded job. This is like extra control point :). If we delete it, then if db is deleted and a request comes at that time, because there wouldnt any record, request would return null. This control point will provide escape this error :)

Reply:
Sorry about that, I did not see the scheduled job. So there is no problem, it is nice to have that control point. Good job :)